src/sql/src/pure.rs

```diff
@@ -1406,48 +1400,34 @@ async fn purify_alter_source(
         Err(PgSourcePurificationError::InsufficientReplicationSlotsAvailable { count: 1 })?;
     }
 
-    let mut publication_tables = mz_postgres_util::publication_info(
+    let new_publication_tables = mz_postgres_util::publication_info(
         &storage_configuration.connection_context.ssh_tunnel_manager,
         &config,
         &pg_source_connection.publication,
     )
     .await?;
 
-    if publication_tables.is_empty() {
+    if new_publication_tables.is_empty() {
         Err(PgSourcePurificationError::EmptyPublication(
             pg_source_connection.publication.to_string(),
         ))?;
     }
 
     let publication_catalog = postgres::derive_catalog_from_publication_tables(
         &pg_connection.database,
-        &publication_tables,
+        &new_publication_tables,
     )?;
 
-    let validated_requested_subsources =
-        subsource_gen(targeted_subsources, &publication_catalog, source_name)?;
-
-    // Determine duplicate references to tables by cross-referencing the table
-    // positions in the current publication info to thei
-    let mut current_subsources = BTreeMap::new();
-    for idx in pg_source_connection.table_casts.keys() {
-        // Table casts all have their values increased by to accommodate for the
-        // primary source--this means that to look them up in the publication
-        // tables you must subtract one.
-        let native_idx = *idx - 1;
-        let table_desc = &pg_source_connection.publication_details.tables[native_idx];
-        current_subsources.insert(
-            UnresolvedItemName(vec![
-                Ident::new(pg_connection.database.clone())?,
-                Ident::new(table_desc.namespace.clone())?,
-                Ident::new(table_desc.name.clone())?,
-            ]),
-            native_idx,
-        );
-    }
+    let unresolved_source_name = UnresolvedItemName::from(source_name.full_item_name().clone());
```

Comment:
how is this different from `outer_source_name` above? Can you leave some comments explaining what's going on with the name mangling?

Reply:
These should have had better variable names--they're different types. One is a `ResolvedItemName`, one is an `UnresolvedItemName`. Amended locally and will push that up in a bit.